File: resize-icons.js
const fs = require('fs');
const path = require('path');

// Simple PNG resizing using canvas (Node.js built-in)
async function resizeIcons() {
  console.log('🔧 Resizing icons for Chrome extension...\n');

  const sharp = require('sharp');
  const sizes = [16, 48, 128];
  const sourceIcon = path.join(__dirname, 'icons', 'icon128.png');
  
  // Check if source exists
  if (!fs.existsSync(sourceIcon)) {
    console.error('❌ Source icon not found: icons/icon128.png');
    console.log('Please ensure you have an icon128.png file in the icons folder.');
    return;
  }

  console.log('📁 Source icon:', sourceIcon);
  console.log('');

  for (const size of sizes) {
    const outputPath = path.join(__dirname, 'icons', `icon${size}.png`);
    
    try {
      await sharp(sourceIcon)
        .resize(size, size, {
          fit: 'contain',
          background: { r: 0, g: 0, b: 0, alpha: 0 }
        })
        .png()
        .toFile(outputPath);
      
      console.log(`✅ Created icon${size}.png (${size}x${size})`);
    } catch (error) {
      console.error(`❌ Failed to create icon${size}.png:`, error.message);
    }
  }

  console.log('\n✨ Icon resizing complete!');
  console.log('📦 Run "npm run build" to rebuild the extension.');
}

// Check if sharp is installed
try {
  require.resolve('sharp');
  resizeIcons();
} catch (e) {
  console.log('📦 Installing sharp package for image processing...\n');
  const { execSync } = require('child_process');
  
  try {
    execSync('npm install sharp --save-dev', { stdio: 'inherit' });
    console.log('\n✅ Sharp installed successfully!');
    console.log('🔄 Run this script again: node resize-icons.js\n');
  } catch (error) {
    console.error('❌ Failed to install sharp. Please install manually:');
    console.log('   npm install sharp --save-dev');
  }
}

File: resize-icons.py
#!/usr/bin/env python3
"""
Resize icons for Chrome extension
Requires: Pillow (pip install Pillow)
"""

import os
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("❌ Pillow not installed. Installing...")
    import subprocess
    subprocess.check_call(["pip", "install", "Pillow"])
    from PIL import Image

def resize_icons():
    print("🔧 Resizing icons for Chrome extension...\n")
    
    icons_dir = Path(__file__).parent / "icons"
    source_icon = icons_dir / "icon128.png"
    
    if not source_icon.exists():
        print(f"❌ Source icon not found: {source_icon}")
        print("Please ensure you have an icon128.png file in the icons folder.")
        return
    
    print(f"📁 Source icon: {source_icon}")
    print("")
    
    sizes = [16, 48, 128]
    
    try:
        # Open the source image
        img = Image.open(source_icon)
        
        # Convert to RGBA if not already
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        for size in sizes:
            output_path = icons_dir / f"icon{size}.png"
            
            # Resize with high-quality resampling
            resized = img.resize((size, size), Image.Resampling.LANCZOS)
            
            # Save as PNG
            resized.save(output_path, 'PNG', optimize=True)
            
            print(f"✅ Created icon{size}.png ({size}x{size})")
        
        print("\n✨ Icon resizing complete!")
        print("📦 Run 'npm run build' to rebuild the extension.")
        
    except Exception as e:
        print(f"❌ Error: {e}")

if __name__ == "__main__":
    resize_icons()
